stack/corepersistence/token/src/main/java/org/apache/usergrid/persistence/token/impl/TokenSerializationImpl.java

```diff
@@ -0,0 +1,451 @@
+            put( "key", DataType.Name.BLOB );
+            put( "column1", DataType.Name.UUID );
+            put( "value", DataType.Name.BLOB ); }};
+    private static final Map<String, String> PRINCIPAL_TOKENS_CLUSTERING_ORDER =
+        new HashMap<String, String>(){{ put( "column1", "ASC" ); }};
+
+
+    private final Session session;
+    private final CassandraConfig cassandraConfig;
+
+
+    @Inject
+    public TokenSerializationImpl(final Session session,
+                                  final CassandraConfig cassandraConfig ) {
+        this.session = session;
+        this.cassandraConfig = cassandraConfig;
+
+    }
+
+
+    @Override
+    public void deleteTokens(final List<UUID> tokenUUIDs, final ByteBuffer principalKeyBuffer){
+
+        Preconditions.checkNotNull(tokenUUIDs, "token UUID list is required");
+        Preconditions.checkNotNull(tokenUUIDs, "principalKeyBuffer is required");
+
+        logger.trace("deleteTokens, token UUIDs: {}", tokenUUIDs);
+
+        final BatchStatement batchStatement = new BatchStatement();
+
+        tokenUUIDs.forEach( tokenUUID ->
+            batchStatement.add(
+                QueryBuilder.delete()
+                    .from(TOKENS_TABLE)
+                    .where(QueryBuilder
+                        .eq("key", DataType.uuid().serialize(tokenUUID, ProtocolVersion.NEWEST_SUPPORTED)))
+            )
+        );
+
+        batchStatement.add(
+            QueryBuilder.delete()
+                .from(PRINCIPAL_TOKENS_TABLE)
+                .where(QueryBuilder
+                    .eq("key", principalKeyBuffer)));
+
+
+        session.execute(batchStatement);
+
+    }
+
+
+    @Override
+    public void revokeToken(final UUID tokenUUID, final ByteBuffer principalKeyBuffer){
+
+        Preconditions.checkNotNull(tokenUUID, "token UUID is required");
+
+        logger.trace("revokeToken, token UUID: {}", tokenUUID);
+
+
+        final BatchStatement batchStatement = new BatchStatement();
+
+        batchStatement.add(
+            QueryBuilder.delete()
+                .from(TOKENS_TABLE)
+                .where(QueryBuilder
+                    .eq("key", DataType.uuid().serialize(tokenUUID, ProtocolVersion.NEWEST_SUPPORTED))));
+
+        if(principalKeyBuffer != null){
+            batchStatement.add(
+                QueryBuilder.delete()
+                    .from(PRINCIPAL_TOKENS_TABLE)
+                    .where(QueryBuilder
+                        .eq("key", principalKeyBuffer))
+                    .and(QueryBuilder
+                        .eq("column1", DataType.uuid().serialize(tokenUUID, ProtocolVersion.NEWEST_SUPPORTED))));
+        }
+
+        session.execute(batchStatement);
+
+    }
+
+
+    @Override
+    public void updateTokenAccessTime(UUID tokenUUID, long accessedTime, long inactiveTime, int ttl ){
+
+        Preconditions.checkNotNull(tokenUUID, "token UUID is required");
+        Preconditions.checkNotNull(accessedTime, "accessedTime is required");
```

Comment:
accessedTime, inactiveTime and ttl are all numbers, shouldn't be a null check here right?